src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -220,13 +220,21 @@ private void CheckInvariants()
             solutionAttributes ??= _solutionAttributes;
             projectIds ??= ProjectIds;
             idToProjectStateMap ??= _projectIdToProjectStateMap;
-            options ??= Options.WithLanguages(GetRemoteSupportedProjectLanguages(idToProjectStateMap));
```

Comment:
❔ Is it possible to make `WithLanguages` and `GetRemoteSupportedProjectLanguages` fast path for the case we are interested in?

Reply:
I think Jason's suggestion is much better, I have implemented that one.